### PR TITLE
Add support for pathlib

### PR DESCRIPTION
**Pull request by bburan**

### CODE REVIEW

File: json_tricks/decoders.py [hunk 1]
```diff
@@ -117,6 +117,23 @@ def noenum_hook(dct):
 	return dct
 
 
+def pathlib_hook(dct):
+		if not isinstance(dct, dict):
```
Comment (mverleg) [inline json_tricks/decoders.py]: This seems to be indented with double tabs, while the rest of the code is single tabs

File: json_tricks/nonp.py [hunk 2]
```diff
@@ -26,9 +26,9 @@
 _cih_instance = ClassInstanceHook()
 _eih_instance = EnumInstanceHook()
 DEFAULT_ENCODERS = [json_date_time_encode, json_complex_encode, json_set_encode,
-                    numeric_types_encode, class_instance_encode, ]
+										numeric_types_encode, class_instance_encode, ]
```
Comment (mverleg) [inline json_tricks/nonp.py]: Something went wrong with the indentation I think

File: tests/test_pathlib.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+This tests timezone-aware date/time objects, which need pytz. Naive date/times should
+work with just Python code functionality, and are tested in `nonp`.
+"""
```
Comment (mverleg) [inline tests/test_pathlib.py]: "This tests Paths, which need pathlib."

File: tests/test_pathlib.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+from pathlib import Path
+from json_tricks import dumps, loads
+from json_tricks.utils import is_py3
+
+
+PATHS = [
+		Path(),
+		Path(__file__),
```
Comment (mverleg) [inline tests/test_pathlib.py]: I think it's better to use a fixed path, so that the test is the same on each system

Maybe different OS paths too?

**Comment by mverleg** [comment]
Great addition, I wasn't aware of pathlib! And thanks for doing the implementation!

The changes are good, follow the setup of other encoders/decoders well.

There are just a few things left to do:

* A few mostly esthetic comments
* Include in readme (and contributors)
* Setup tox tests for pathlib

I could have a look this weekend to do the finishing touches, if you don't have time. I appreciate that you already did almost all the work :+1:

File: json_tricks/decoders.py (hunk 1 above)
Comment (bburan) [reply json_tricks/decoders.py]: Sorry! I didn't realize the file was formatted with tabs when I first started editing, so I had to go back and convert everything from spaces to tabs. I'll fix this.

File: tests/test_pathlib.py (hunk 4 above)
Comment (bburan) [reply tests/test_pathlib.py]: Done.

File: tests/test_pathlib.py (hunk 3 above)
Comment (bburan) [reply tests/test_pathlib.py]: Done.

**Comment by bburan** [comment]
I'm not an expert on tox, so I wasn't sure what to do. I updated the pathlib test to hide it inside an if `is_py3` block, but it looks like tox is smart enough to only run certain tests depending on the Python version? I think I addressed all the other comments you have, so please let me know if there's anything else. Thanks for taking the time to review this!

**Comment by mverleg** [comment]
Okay looks great! I'll have a look at tox after merging, and add you to the contributors. Thanks for the PR!

**Comment by mverleg** [comment]
This feature is available in the latest version through pip

**Comment by bburan** [comment]
Thanks!

**Pull request merged**